### PR TITLE
Do not violate NodeGroup's MinSize when "fixing" NodeGroupSize

### PR DESCRIPTION
**Pull request by zuzzas**

### CODE REVIEW

**Comment by hardikdr** [comment]
Thanks @zuzzas, this is a much-needed improvement. 

/lgtm

**Comment by hardikdr** [comment]
Sorry, I missed it's a draft yet.

**Comment by zuzzas** [comment]
> Sorry, I missed it's a draft yet. 

Haven't written tests yet. Soon!

File: cluster-autoscaler/core/static_autoscaler.go [hunk 1]
```diff
@@ -565,7 +565,9 @@ func fixNodeGroupSize(context *context.AutoscalingContext, clusterStateRegistry
 		}
 		if incorrectSize.FirstObserved.Add(context.MaxNodeProvisionTime).Before(currentTime) {
 			delta := incorrectSize.CurrentSize - incorrectSize.ExpectedSize
-			if delta < 0 {
+			if (nodeGroup.MinSize() + delta) < nodeGroup.MinSize() {
```
Comment (mwielgus) [inline cluster-autoscaler/core/static_autoscaler.go:568]: ng.minsize + delta < ng.minsize is equivalent to delta < 0 a cople lines below

**Pull request closed without merging**